### PR TITLE
Assembly stats

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Some assembly stats added to the multiqc report!
http://ocimum.science.ru.nl/multiqc_GRCz11.html#assembly_stats

**Checklist**
- [x] I made a PR to develop (not master)
- [ ] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

File: seq2science/scripts/assembly_stats.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+img = io.BytesIO()
+fig.savefig(img, format='png')
+img.seek(0)
+html = '<img src="data:image/png;base64, {}">'.format(base64.b64encode(img.getvalue()).decode('utf-8'))
+
+# if
+if hasattr(snakemake.input, "annotation"):
+    with open(snakemake.input.annotation) as f:
+        annotation = f.read()
+    nr_genes = annotation.count("transcript\t")
```
[the remark this answers is not in the document]
Comment (siebrenf) [reply seq2science/scripts/assembly_stats.py:66]: The number of unique gene_ids in this particular gtf is 32520, which is the same as when I use 
```
with open(gene_gtf) as f:
    annotation = f.read()
annotation.count("gene\t")
```

**Review by siebrenf — APPROVED** [review]
one hardcoded path and transcripts are called genes in your output (fix suggested).

File: seq2science/scripts/assembly_stats.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+outfile = snakemake.output[0]
+assembly = snakemake.wildcards.assembly
+
+# empty old file (if exists)
+open(outfile, "w").close()
+
+# read the assembly
+fa = pyfaidx.Fasta("/vol/genomes/oryLat2/oryLat2.fa")
```
[the remark this answers is not in the document]
Comment (Maarten-vd-Sande) [reply seq2science/scripts/assembly_stats.py:17]: :see_no_evil: damn!

File: seq2science/scripts/assembly_stats.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+# now sort the sizes from large to small
+sizes = list(sorted(sizes, reverse=True))
+
+# get the cumulative and total sizes
+cum_sizes = list(accumulate(sizes))
+total_size = sum(sizes)
+
+# get the n50 and l50 stats
+n50_done = n75_done = False
```
[the remark this answers is not in the document]
Comment (Maarten-vd-Sande) [reply seq2science/scripts/assembly_stats.py:38]: Only with mutable objects. This is like the most confusing and worst design choices of python. But e.g. True and Falses, and numbers are not mutable objects, so you can do that safely here.

File: seq2science/scripts/assembly_stats.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maarten-vd-Sande) [reply seq2science/scripts/assembly_stats.py]: Ah cool, so what's the difference? A single gene can have multiple transcripts (differentt start/stop and/or alternative splicings?)

**Comment by Maarten-vd-Sande** [comment]
Thanks for the review @siebrenf , that was a sloppy one :see_no_evil:

**Pull request merged**